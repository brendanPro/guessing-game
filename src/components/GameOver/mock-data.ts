import type { Pokemon } from "@/types/pokemon";

export const mockPikachu: Pokemon = {
    id: 25,
    name: "pikachu",
    frenchName: "Pikachu",
    sprites: {
      front_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
      back_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png",
      front_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/25.png",
      back_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/25.png",
    },
    types: [
      {
        type: {
          name: "electric",
        },
      },
    ],
    height: 4,
    weight: 60,
    stats: [
      {
        base_stat: 35,
        stat: {
          name: "hp",
        },
      },
    ],
  };
  
export const mockCharizard: Pokemon = {
    id: 6,
    name: "charizard",
    frenchName: "Dracaufeu",
    sprites: {
      front_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png",
      back_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png",
      front_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png",
      back_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png",
    },
    types: [
      {
        type: {
          name: "fire",
        },
      },
      {
        type: {
          name: "flying",
        },
      },
    ],
    height: 17,
    weight: 905,
    stats: [
      {
        base_stat: 78,
        stat: {
          name: "hp",
        },
      },
    ],
  };


export const mockMewtwo: Pokemon = {
  id: 150,
  name: 'mewtwo',
  frenchName: 'Mewtwo',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/150.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/150.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/150.png',
  },
  types: [
    {
      type: {
        name: 'psychic',
      },
    },
  ],
  height: 20,
  weight: 1220,
  stats: [
    {
      base_stat: 106,
      stat: {
        name: 'hp',
      },
    },
  ],
};