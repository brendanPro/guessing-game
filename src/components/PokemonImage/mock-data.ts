import type { Pokemon } from '@/types/pokemon';

export const mockPikachu: Pokemon = {
  id: 25,
  name: 'pikachu',
  frenchName: 'Pikachu',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/25.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/25.png',
  },
  types: [
    {
      type: {
        name: 'electric',
      },
    },
  ],
  height: 4,
  weight: 60,
  stats: [
    {
      base_stat: 35,
      stat: {
        name: 'hp',
      },
    },
  ],
};

export const mockCharizard: Pokemon = {
  id: 6,
  name: 'charizard',
  frenchName: 'Dracaufeu',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png',
  },
  types: [
    {
      type: {
        name: 'fire',
      },
    },
    {
      type: {
        name: 'flying',
      },
    },
  ],
  height: 17,
  weight: 905,
  stats: [
    {
      base_stat: 78,
      stat: {
        name: 'hp',
      },
    },
  ],
};

export const mockMewtwo: Pokemon = {
  id: 150,
  name: 'mewtwo',
  frenchName: 'Mewtwo',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/150.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/150.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/150.png',
  },
  types: [
    {
      type: {
        name: 'psychic',
      },
    },
  ],
  height: 20,
  weight: 1220,
  stats: [
    {
      base_stat: 106,
      stat: {
        name: 'hp',
      },
    },
  ],
};

export const mockGyarados: Pokemon = {
  id: 130,
  name: 'gyarados',
  frenchName: 'LÃ©viator',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/130.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/130.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/130.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/130.png',
  },
  types: [
    {
      type: {
        name: 'water',
      },
    },
    {
      type: {
        name: 'flying',
      },
    },
  ],
  height: 65,
  weight: 2350,
  stats: [
    {
      base_stat: 95,
      stat: {
        name: 'hp',
      },
    },
  ],
};

