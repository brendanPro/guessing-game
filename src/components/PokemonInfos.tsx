import { Card, CardContent, CardHeader, CardTitle } from "./ui/card";
import type { Pokemon } from "@/types/pokemon";

interface PokemonInfosProps {
    pokemon: Pokemon;
}

export function PokemonInfos({pokemon}: PokemonInfosProps) {
    return (
        <Card>
          <CardHeader>
            <CardTitle className="text-center">Pokemon Information</CardTitle>
          </CardHeader>
          <CardContent>
            <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
              <div>
                <h4 className="font-semibold mb-2">Basic Info</h4>
                <p><strong>Name:</strong> {pokemon.frenchName}</p>
                <p><strong>Height:</strong> {pokemon.height / 10}m</p>
                <p><strong>Weight:</strong> {pokemon.weight / 10}kg</p>
              </div>
              <div>
                <h4 className="font-semibold mb-2">Types</h4>
                <div className="flex gap-2">
                  {pokemon.types.map((type, index) => (
                    <span 
                      key={index}
                      className="px-2 py-1 bg-primary text-primary-foreground rounded text-sm"
                    >
                      {type.type.name}
                    </span>
                  ))}
                </div>
              </div>
            </div>
          </CardContent>
        </Card>
    )
}