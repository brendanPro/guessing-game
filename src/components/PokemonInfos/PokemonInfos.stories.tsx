import "@/index.css"
import type { Meta, StoryObj } from '@storybook/react';
import { PokemonInfos } from './PokemonInfos';
import { mockCharizard, mockGyarados, mockMewtwo, mockPikachu } from "./mock-data";


const meta = {
  title: 'Game/PokemonInfos',
  component: PokemonInfos,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    pokemon: {
      control: 'object',
      description: 'The Pokemon to display information about',
    },
  },
} satisfies Meta<typeof PokemonInfos>;

export default meta;
type Story = StoryObj<typeof meta>;

export const WithPikachu: Story = {
  args: {
    pokemon: mockPikachu,
  },
};

export const WithCharizard: Story = {
  args: {
    pokemon: mockCharizard,
  },
};

export const WithMewtwo: Story = {
  args: {
    pokemon: mockMewtwo,
  },
};

export const WithGyarados: Story = {
  args: {
    pokemon: mockGyarados,
  },
};

export const WithSingleType: Story = {
  args: {
    pokemon: mockPikachu,
  },
};

export const WithDualType: Story = {
  args: {
    pokemon: mockCharizard,
  },
};

export const WithLongName: Story = {
  args: {
    pokemon: {
      ...mockCharizard,
      frenchName: 'Tortank-MÃ©ga-X',
    },
  },
};

export const WithHeavyPokemon: Story = {
  args: {
    pokemon: mockGyarados,
  },
};

export const WithLightPokemon: Story = {
  args: {
    pokemon: {
      ...mockPikachu,
      height: 2,
      weight: 30,
    },
  },
};

