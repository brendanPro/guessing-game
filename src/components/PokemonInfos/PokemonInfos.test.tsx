import { describe, test, expect } from "bun:test";
import { renderToStaticMarkup } from "react-dom/server";
import { PokemonInfos } from "./PokemonInfos";
import { mockCharizard, mockPikachu, mockMewtwo, mockGyarados } from './mock-data';


describe("PokemonInfos Component", () => {
  test("snapshot with Pikachu", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Charizard (dual type)", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockCharizard} />);
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Mewtwo", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockMewtwo} />);
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Gyarados (heavy pokemon)", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockGyarados} />);
    expect(html).toMatchSnapshot();
  });

  test("contains Pokemon Information title", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("Pokemon Information");
  });

  test("displays pokemon french name", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("Pikachu");
  });

  test("displays correct french name for Charizard", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockCharizard} />);
    expect(html).toContain("Dracaufeu");
  });

  test("displays height in meters", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("0.4m"); // height 4 / 10
  });

  test("displays weight in kilograms", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("6kg"); // weight 60 / 10
  });

  test("displays single type correctly", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("electric");
  });

  test("displays dual types correctly", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockCharizard} />);
    expect(html).toContain("fire");
    expect(html).toContain("flying");
  });

  test("has correct structure with Basic Info section", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("Basic Info");
  });

  test("has correct structure with Types section", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("Types");
  });

  test("displays large pokemon measurements correctly", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockGyarados} />);
    expect(html).toContain("6.5m"); // height 65 / 10
    expect(html).toContain("235kg"); // weight 2350 / 10
  });

  test("contains Name label", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("<strong>Name:</strong>");
  });

  test("contains Height label", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("<strong>Height:</strong>");
  });

  test("contains Weight label", () => {
    const html = renderToStaticMarkup(<PokemonInfos pokemon={mockPikachu} />);
    expect(html).toContain("<strong>Weight:</strong>");
  });
});

