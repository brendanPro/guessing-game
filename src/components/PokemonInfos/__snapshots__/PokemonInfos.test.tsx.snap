// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonInfos Component snapshot with Pikachu 1`] = `"<div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm"><div data-slot="card-header" class="flex flex-col gap-1.5 p-6"><div data-slot="card-title" class="leading-none font-semibold tracking-tight text-center">Pokemon Information</div></div><div data-slot="card-content" class="p-6 pt-0"><div class="grid grid-cols-1 md:grid-cols-2 gap-4"><div><h4 class="font-semibold mb-2">Basic Info</h4><p><strong>Name:</strong> Pikachu</p><p><strong>Height:</strong> 0.4m</p><p><strong>Weight:</strong> 6kg</p></div><div><h4 class="font-semibold mb-2">Types</h4><div class="flex gap-2"><span class="px-2 py-1 bg-primary text-primary-foreground rounded text-sm">electric</span></div></div></div></div></div>"`;

exports[`PokemonInfos Component snapshot with Charizard (dual type) 1`] = `"<div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm"><div data-slot="card-header" class="flex flex-col gap-1.5 p-6"><div data-slot="card-title" class="leading-none font-semibold tracking-tight text-center">Pokemon Information</div></div><div data-slot="card-content" class="p-6 pt-0"><div class="grid grid-cols-1 md:grid-cols-2 gap-4"><div><h4 class="font-semibold mb-2">Basic Info</h4><p><strong>Name:</strong> Dracaufeu</p><p><strong>Height:</strong> 1.7m</p><p><strong>Weight:</strong> 90.5kg</p></div><div><h4 class="font-semibold mb-2">Types</h4><div class="flex gap-2"><span class="px-2 py-1 bg-primary text-primary-foreground rounded text-sm">fire</span><span class="px-2 py-1 bg-primary text-primary-foreground rounded text-sm">flying</span></div></div></div></div></div>"`;

exports[`PokemonInfos Component snapshot with Mewtwo 1`] = `"<div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm"><div data-slot="card-header" class="flex flex-col gap-1.5 p-6"><div data-slot="card-title" class="leading-none font-semibold tracking-tight text-center">Pokemon Information</div></div><div data-slot="card-content" class="p-6 pt-0"><div class="grid grid-cols-1 md:grid-cols-2 gap-4"><div><h4 class="font-semibold mb-2">Basic Info</h4><p><strong>Name:</strong> Mewtwo</p><p><strong>Height:</strong> 2m</p><p><strong>Weight:</strong> 122kg</p></div><div><h4 class="font-semibold mb-2">Types</h4><div class="flex gap-2"><span class="px-2 py-1 bg-primary text-primary-foreground rounded text-sm">psychic</span></div></div></div></div></div>"`;

exports[`PokemonInfos Component snapshot with Gyarados (heavy pokemon) 1`] = `"<div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm"><div data-slot="card-header" class="flex flex-col gap-1.5 p-6"><div data-slot="card-title" class="leading-none font-semibold tracking-tight text-center">Pokemon Information</div></div><div data-slot="card-content" class="p-6 pt-0"><div class="grid grid-cols-1 md:grid-cols-2 gap-4"><div><h4 class="font-semibold mb-2">Basic Info</h4><p><strong>Name:</strong> LÃ©viator</p><p><strong>Height:</strong> 6.5m</p><p><strong>Weight:</strong> 235kg</p></div><div><h4 class="font-semibold mb-2">Types</h4><div class="flex gap-2"><span class="px-2 py-1 bg-primary text-primary-foreground rounded text-sm">water</span><span class="px-2 py-1 bg-primary text-primary-foreground rounded text-sm">flying</span></div></div></div></div></div>"`;
